vlasiator.cpp

```diff
@@ -478,7 +479,7 @@ int main(int argn,char* args[]) {
    phiprof::stop("Init DROs");  
    
    // Free up memory:
-   readparameters.finalize();
+   readparameters.~Readparameters();
```

Comment:
Uh-oh, since readparameters is an auto variable, this explicit call of th destructor is a bit dangerous.
Especially since it calls unprotected ```delete``` on some pointers in there. Won't this cause a double free, when th readparameters object is eventually deconstructed at the end of main?
Not that it would really matter if it crashes at the very end, but I would suggest putting a nullpointer check around the deletes in the destructor.

Reply:
This is inherited from the original struct of `Readparameters`, but I agree it is dangerous. I kept it here as it was because maybe freeing the memory manually here would save some memory in the actual time marching loops. 

One thing I noticed in this destructor
```c++
Readparameters::~Readparameters() {
   delete descriptions;
   delete variables;
   descriptions = NULL;
   variables = NULL;
}
```
was that without setting these two pointers to `NULL`, the program will crash in the end. I am not sure why. Will there be an enforced auto free at the end of `main()`? What should I do to make it better?

Reply:
The readparameters object itself will fall out of scope at the end of main, so the destructor will run again in that case. If you don't set them to NULL, that will be a certain double free. This probably causes the crash.

We could either guard the deletes:
```c++
if(descriptions != nullptr) {
   delete descriptions;
}
```
or we decide that we don't care about this little bit of wasted memory in a simulation that's multiple gigabytes heavy anyway.